clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -155,7 +155,10 @@ public void run() {
             this.accumulator.abortIncompleteBatches();
         }
         try {
-            this.client.close();
+            if (!forceClose)
+                this.client.closeGracefully();
```

Comment:
Does this actually buy us anything here? If `forceClose` is false, then doesn't that mean that there are no more in-flight requests? I think we still increment the in-flight request count even if the client doesn't expect a response.